core/kazoo_voicemail/src/kvm_migrate.erl

```diff
@@ -0,0 +1,372 @@
+    lager:info("voicemail migration finished, going down"),
+    {'stop', 'normal', State};
+handle_info({'timeout', Ref, _Msg}, #state{timer_ref = Ref
+                                          ,account_ids = [AccountId | AccountIds]
+                                          ,read_offset = Offset
+                                          }=State) ->
+    case crawl_account(AccountId, Offset) of
+        {'ok', NewOffset} ->
+            {'noreply', State#state{timer_ref = cleanup_mailbox_timer()
+                                   ,read_offset = NewOffset
+                                   }};
+        _ ->
+            lager:info("voicemail migration for account ~s is finished, ~b accounts remain to process"
+                      ,[AccountId, length(AccountIds)]),
+            {'noreply', State#state{timer_ref = cleanup_account_timer()
+                                   ,account_ids = AccountIds
+                                   ,read_offset = 0
+                                   }}
+    end;
+handle_info(_Info, State) ->
+    lager:debug("unhandled msg: ~p", [_Info]),
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function is called by a gen_server when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_server terminates
+%% with Reason. The return value is ignored.
+%% @end
+%%--------------------------------------------------------------------
+-spec terminate(any(), state()) -> 'ok'.
+terminate(_Reason, _State) ->
+    lager:debug("~s terminating: ~p", [?SERVER, _Reason]).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Convert process state when code is changed
+%% @end
+%%--------------------------------------------------------------------
+-spec code_change(any(), state(), any()) -> {'ok', state()}.
+code_change(_OldVsn, State, _Extra) ->
+    {'ok', State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec crawl_account(ne_binary(), api_integer()) -> 'done' | db_ret().
+crawl_account(_AccountId, Offset) when Offset < 0 -> 'done';
+crawl_account(AccountId, Offset) ->
+    lager:info("crawling account ~s with offset ~p", [AccountId, Offset]),
+
+    ViewOpts = props:filter_undefined(
+                 [{'limit', ?MAX_BOX_PROCESS}
+                 ,{'skip', Offset}
+                 ,'include_docs'
+                 ]),
+    case get_account_vmboxes(AccountId, ViewOpts) of
+        {'ok', Results} ->
+            do_migrate(AccountId, Results),
+            {'ok', length(Results) - Offset};
```

Comment:
Offset can be undefined (as per spec), and first clause does not rule it out: `undefined < 0 =:= false`.
This will give badarith